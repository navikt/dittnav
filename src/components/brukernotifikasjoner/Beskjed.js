import React from 'react';
import { useLocation } from 'react-router-dom';
import { bool } from 'prop-types';
import useSikkerhetsnivaa from '../../hooks/useSikkerhetsnivaa';
import useStore from '../../hooks/useStore';
import { transformTolokalDatoTid } from '../../utils/datoUtils';
import PanelMedIkon from '../common/PanelMedIkon';
import { postDigisosDone, postDone, tokenExpiresSoon } from '../../Api';
import IkonBeskjed from '../../assets/IkonBeskjed';
import InnloggingsstatusType from '../../types/InnloggingsstatusType';
import BeskjedType from '../../types/BeskjedType';
import { GoogleAnalyticsAction, GoogleAnalyticsCategory, trackEvent } from '../../utils/googleAnalytics';

const remove = (beskjed, removeBeskjed, visInnloggingsModal) => {
  if (beskjed.produsent === 'digiSos') {
    postDigisosDone({
      eventId: beskjed.eventId,
      grupperingsId: beskjed.grupperingsId,
      uid: beskjed.uid,
    });
  }
  postDone({
    eventId: beskjed.eventId,
    uid: beskjed.uid,
  }).then((headers) => {
    if (tokenExpiresSoon(headers)) {
      visInnloggingsModal();
    }
  });
  removeBeskjed(beskjed);
};

const addTilInaktiveHvisErAktiv = (beskjed, addInaktivBeskjed, erAktiv) => {
  if (erAktiv) {
    addInaktivBeskjed(beskjed);
  }
};

const onClickBeskjed = (beskjed, removeBeskjed, addInaktivBeskjed, visInnloggingsModal, erAktiv) => {
  remove(beskjed, removeBeskjed, visInnloggingsModal);
  addTilInaktiveHvisErAktiv(beskjed, addInaktivBeskjed, erAktiv);
  trackEvent(GoogleAnalyticsCategory.Forside, GoogleAnalyticsAction.BeskjedLukk, '');
};

const Beskjed = ({ beskjed, innloggingsstatus, erAktiv, erInaktiv }) => {
  const location = useLocation();
  const { removeBeskjed, addInaktivBeskjed, visInnloggingsModal } = useStore();

  const sikkerhetsnivaa = useSikkerhetsnivaa(beskjed, 'beskjed', innloggingsstatus);
  const lenkeTekst = sikkerhetsnivaa.skalMaskeres ? 'beskjed.lenke.stepup.tekst' : 'beskjed.lenke.tekst';
  const lokalDatoTid = transformTolokalDatoTid(beskjed.eventTidspunkt);

  const visKnapp = !(sikkerhetsnivaa.skalMaskeres || erInaktiv);

  return (
    <PanelMedIkon
      className="beskjed"
      alt="Beskjed"
      overskrift={sikkerhetsnivaa.tekst}
      etikett={lokalDatoTid}
      onClick={() => onClickBeskjed(beskjed, removeBeskjed, addInaktivBeskjed, visInnloggingsModal, erAktiv)}
      skjermleserTekst="beskjed.knapp.skjermleser.tekst"
      lenke={sikkerhetsnivaa.lenke}
      lenkeTekst={lenkeTekst}
      gaCategory={`Ditt NAV${location.pathname}`}
      gaAction={GoogleAnalyticsAction.Beskjed}
      knapp={visKnapp}
    >
      <IkonBeskjed />
    </PanelMedIkon>
  );
};

Beskjed.propTypes = {
  beskjed: BeskjedType,
  innloggingsstatus: InnloggingsstatusType,
  erAktiv: bool,
  erInaktiv: bool,
};

Beskjed.defaultProps = {
  beskjed: null,
  innloggingsstatus: null,
  erAktiv: false,
  erInaktiv: false,
};

export default Beskjed;
