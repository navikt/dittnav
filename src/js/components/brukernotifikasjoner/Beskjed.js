import React, { useEffect } from 'react';
import { bool } from 'prop-types';
import useSikkerhetsnivaa from '../../hooks/useSikkerhetsnivaa';
import useBeskjedStore from '../../hooks/useBeskjedStore';
import { hotjarTrigger, hotjarSafetyStub } from '../../utils/Hotjar';
import transformTolokalDatoTid from '../../utils/DatoUtils';
import PanelMedIkon from '../common/PanelMedIkon';
import IkonBeskjed from '../../../assets/IkonBeskjed';
import { REMOVE_BESKJED, ADD_INAKTIV_BESKJED } from '../../types/Actions';
import InnloggingType from '../../types/InnloggingType';
import BeskjedType from '../../types/BeskjedType';

const remove = (beskjed, dispatch) => dispatch({
  type: REMOVE_BESKJED,
  payload: beskjed,
});

const addInaktiv = (beskjed, dispatch) => dispatch({
  type: ADD_INAKTIV_BESKJED,
  payload: beskjed,
});

const onClickBeskjed = (beskjed, dispatch, erAktiv) => {
  remove(beskjed, dispatch);
  hotjarTrigger('beskjed_trykket_ok');

  if (erAktiv) {
    addInaktiv(beskjed, dispatch);
  }
};

const Beskjed = ({ beskjed, innlogging, erAktiv, erInaktiv }) => {
  const { dispatch } = useBeskjedStore();
  const sikkerhetsnivaa = useSikkerhetsnivaa(beskjed, 'beskjed', innlogging);

  useEffect(() => {
    hotjarSafetyStub();
  }, []);

  const lenkeTekst = sikkerhetsnivaa.skalMaskeres ? 'beskjed.lenke.stepup.tekst' : 'beskjed.lenke.tekst';
  const lokalDatoTid = transformTolokalDatoTid(beskjed.eventTidspunkt);

  const visKnapp = !(sikkerhetsnivaa.skalMaskeres || erInaktiv)

  return (
    <PanelMedIkon
      className="beskjed"
      data-ga="Dittnav/Varsel"
      alt="Beskjed"
      overskrift={sikkerhetsnivaa.tekst}
      etikett={lokalDatoTid}
      onClick={() => onClickBeskjed(beskjed, dispatch, erAktiv)}
      skjermleserTekst="beskjed.knapp.skjermleser.tekst"
      lenke={sikkerhetsnivaa.lenke}
      lenkeTekst={lenkeTekst}
      knapp={visKnapp}
    >
      <IkonBeskjed />
    </PanelMedIkon>
  );
};

Beskjed.propTypes = {
  beskjed: BeskjedType,
  innlogging: InnloggingType,
  erAktiv: bool,
  erInaktiv: bool,
};

Beskjed.defaultProps = {
  beskjed: null,
  innlogging: null,
  erAktiv: false,
  erInaktiv: false,
};

export default Beskjed;
