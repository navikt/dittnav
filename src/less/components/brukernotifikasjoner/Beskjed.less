.beskjed {
  margin-bottom: 1em;
  padding: 0.75em 0.75em !important;
  display: flex !important;
  flex-direction: row;
  background-color: #f5fcfd !important;
  border: 1px solid @navLysBlaLighten40 !important;

  .panel__innhold {
    padding: 0;
  }

  &__ikon {
    align-self: center;

    svg {
      vertical-align: middle;
    }
  }

  &__tekst {
    align-self: center;
    padding-left: 1em;
  }

  &__etikett {
    color: @navGra60;
  }

  &__knapp {
    align-self: center;
    margin-left: auto;
    padding-left: 0.25em;
    padding-right: 0.5em;
  }
}

@media screen and (max-width: @mobile-breakpoint) {
  .beskjed {
    padding: 1.025em 1.025em !important;
    padding-top: 1.25em !important;
    padding-bottom: 1.25em !important;

    &__ikon {
      display: none;
    }

    &__tekst {
      padding-left: 0;
    }

    &__knapp {
      padding-left: 0.725em;
      padding-right: 0;
    }
  }
}
