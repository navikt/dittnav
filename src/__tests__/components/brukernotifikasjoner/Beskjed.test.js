import * as React from 'react';
const ReactTestRenderer = require('react-test-renderer');
import Beskjed from 'js/components/brukernotifikasjoner/Beskjed';
import wrapIntl from 'js/IntlTestHelper';

test('Beskjed with link', () => {
  const beskjed = {
    uid: '2313ef1a-a69a-45d9-a3ff-38e4522710e0',
    eventTidspunkt: '2020-03-13T08:53:17.47Z',
    eventId: '1584093197470',
    tekst: 'Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepenger.',
    link: 'https://enNyLenke',
    sistOppdatert: '2020-03-13T08:53:17.773555Z',
    sikkerhetsnivaa: 4,
  };

  const innlogging = {
    securityLevel: '4',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Beskjed beskjed={beskjed} beskjeder={null} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Beskjed with empty link', () => {
  const beskjed = {
    uid: '2313ef1a-a69a-45d9-a3ff-38e4522710e0',
    eventTidspunkt: '2020-03-13T08:53:17.47Z',
    eventId: '1584093197470',
    tekst: 'Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepenger.',
    link: '',
    sistOppdatert: '2020-03-13T08:53:17.773555Z',
    sikkerhetsnivaa: 4,
  };

  const innlogging = {
    securityLevel: '4',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Beskjed beskjed={beskjed} beskjeder={null} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Beskjed with null as link', () => {
  const beskjed = {
    uid: '2313ef1a-a69a-45d9-a3ff-38e4522710e0',
    eventTidspunkt: '2020-03-13T08:53:17.47Z',
    eventId: '1584093197470',
    tekst: 'Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepenger.',
    link: null,
    sistOppdatert: '2020-03-13T08:53:17.773555Z',
    sikkerhetsnivaa: 4,
  };

  const innlogging = {
    securityLevel: '4',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Beskjed beskjed={beskjed} beskjeder={null} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Beskjed with sikkerhetsnivaa 4 and innloggingsnivaa 4', () => {
  const beskjed = {
    uid: '2313ef1a-a69a-45d9-a3ff-38e4522710e0',
    eventTidspunkt: '2020-03-13T08:53:17.47Z',
    eventId: '1584093197470',
    tekst: 'Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepenger.',
    link: 'https://enNyLenke',
    sistOppdatert: '2020-03-13T08:53:17.773555Z',
    sikkerhetsnivaa: 4,
  };

  const innlogging = {
    securityLevel: '4',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Beskjed beskjed={beskjed} beskjeder={null} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Beskjed with sikkerhetsnivaa 4 and innloggingsnivaa 3', () => {
  const beskjed = {
    uid: '2313ef1a-a69a-45d9-a3ff-38e4522710e0',
    eventTidspunkt: '2020-03-13T08:53:17.47Z',
    eventId: '1584093197470',
    tekst: 'Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepenger.',
    link: 'https://enNyLenke',
    sistOppdatert: '2020-03-13T08:53:17.773555Z',
    sikkerhetsnivaa: 4,
  };

  const innlogging = {
    securityLevel: '3',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Beskjed beskjed={beskjed} beskjeder={null} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Beskjed with sikkerhetsnivaa 3 and innloggingsnivaa 3', () => {
  const beskjed = {
    uid: '2313ef1a-a69a-45d9-a3ff-38e4522710e0',
    eventTidspunkt: '2020-03-13T08:53:17.47Z',
    eventId: '1584093197470',
    tekst: 'Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepenger.',
    link: 'https://enNyLenke',
    sistOppdatert: '2020-03-13T08:53:17.773555Z',
    sikkerhetsnivaa: 4,
  };

  const innlogging = {
    securityLevel: '3',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Beskjed beskjed={beskjed} beskjeder={null} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Beskjed with sikkerhetsnivaa 3 and innloggingsnivaa 4', () => {
  const beskjed = {
    uid: '2313ef1a-a69a-45d9-a3ff-38e4522710e0',
    eventTidspunkt: '2020-03-13T08:53:17.47Z',
    eventId: '1584093197470',
    tekst: 'Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepenger.',
    link: 'https://enNyLenke',
    sistOppdatert: '2020-03-13T08:53:17.773555Z',
    sikkerhetsnivaa: 4,
  };

  const innlogging = {
    securityLevel: '4',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Beskjed beskjed={beskjed} beskjeder={null} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});
