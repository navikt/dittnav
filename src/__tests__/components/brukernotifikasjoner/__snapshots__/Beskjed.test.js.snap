// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beskjed with empty link 1`] = `
<div
  className="panel beskjed panel--border"
>
  <div
    className="beskjed__ikon"
  >
    <svg
      fill="none"
      height="48"
      viewBox="0 0 54 54"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="27"
        cy="27"
        fill="#97E6FF"
        r="27"
      />
      <path
        clipRule="evenodd"
        d="M15 27C15 20.3833 20.3833 15 27 15C33.6167 15 39 20.3833 39 27C39 33.6167 33.6167 39 27 39C20.3833 39 15 33.6167 15 27ZM16.0435 27C16.0435 33.0417 20.9583 37.9565 27 37.9565C33.0417 37.9565 37.9565 33.0417 37.9565 27C37.9565 20.9582 33.0417 16.0435 27 16.0435C20.9583 16.0435 16.0435 20.9582 16.0435 27ZM30.1304 33.7824H27.5217V24.9129C27.5217 24.6249 27.288 24.3911 27 24.3911H24.913C24.625 24.3911 24.3913 24.6249 24.3913 24.9129C24.3913 25.2009 24.625 25.4346 24.913 25.4346H26.4783V33.7824H23.8696C23.5816 33.7824 23.3478 34.0162 23.3478 34.3042C23.3478 34.5922 23.5816 34.8259 23.8696 34.8259H30.1304C30.4184 34.8259 30.6522 34.5922 30.6522 34.3042C30.6522 34.0162 30.4184 33.7824 30.1304 33.7824ZM27.5217 20.7393C27.5217 21.3156 27.0546 21.7828 26.4783 21.7828C25.902 21.7828 25.4348 21.3156 25.4348 20.7393C25.4348 20.163 25.902 19.6958 26.4783 19.6958C27.0546 19.6958 27.5217 20.163 27.5217 20.7393Z"
        fill="#262626"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="beskjed__tekst"
  >
    <p
      className="typo-normal"
    >
      <span>
        Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepen... 
        
        <span>
          
          <a
            className="lenke utvid-tekst-lenke__aapne"
            href="#"
            onClick={[Function]}
          >
            Åpne
            <span
              className="nav-frontend-chevron chevronboks chevron--ned"
            />
          </a>
        </span>
      </span>
    </p>
    <p
      className="typo-undertekst beskjed__etikett"
    >
      13.03.2020 - 09:53
    </p>
  </div>
  <div
    className="beskjed__knapp"
  >
    <button
      className="lukknapp lukknapp--bla"
      onClick={[Function]}
    >
      <span
        className="text-hide"
      >
        Fjern beskjeden
      </span>
    </button>
  </div>
</div>
`;

exports[`Beskjed with link 1`] = `
<div
  className="panel beskjed panel--border"
>
  <div
    className="beskjed__ikon"
  >
    <svg
      fill="none"
      height="48"
      viewBox="0 0 54 54"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="27"
        cy="27"
        fill="#97E6FF"
        r="27"
      />
      <path
        clipRule="evenodd"
        d="M15 27C15 20.3833 20.3833 15 27 15C33.6167 15 39 20.3833 39 27C39 33.6167 33.6167 39 27 39C20.3833 39 15 33.6167 15 27ZM16.0435 27C16.0435 33.0417 20.9583 37.9565 27 37.9565C33.0417 37.9565 37.9565 33.0417 37.9565 27C37.9565 20.9582 33.0417 16.0435 27 16.0435C20.9583 16.0435 16.0435 20.9582 16.0435 27ZM30.1304 33.7824H27.5217V24.9129C27.5217 24.6249 27.288 24.3911 27 24.3911H24.913C24.625 24.3911 24.3913 24.6249 24.3913 24.9129C24.3913 25.2009 24.625 25.4346 24.913 25.4346H26.4783V33.7824H23.8696C23.5816 33.7824 23.3478 34.0162 23.3478 34.3042C23.3478 34.5922 23.5816 34.8259 23.8696 34.8259H30.1304C30.4184 34.8259 30.6522 34.5922 30.6522 34.3042C30.6522 34.0162 30.4184 33.7824 30.1304 33.7824ZM27.5217 20.7393C27.5217 21.3156 27.0546 21.7828 26.4783 21.7828C25.902 21.7828 25.4348 21.3156 25.4348 20.7393C25.4348 20.163 25.902 19.6958 26.4783 19.6958C27.0546 19.6958 27.5217 20.163 27.5217 20.7393Z"
        fill="#262626"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="beskjed__tekst"
  >
    <p
      className="typo-normal"
    >
      <span>
        Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepen... 
        
        <span>
          
          <a
            className="lenke utvid-tekst-lenke__aapne"
            href="#"
            onClick={[Function]}
          >
            Åpne
            <span
              className="nav-frontend-chevron chevronboks chevron--ned"
            />
          </a>
        </span>
      </span>
    </p>
    <p
      className="typo-undertekst beskjed__etikett"
    >
      13.03.2020 - 09:53
    </p>
  </div>
  <div
    className="beskjed__knapp"
  >
    <button
      className="lukknapp lukknapp--bla"
      onClick={[Function]}
    >
      <span
        className="text-hide"
      >
        Fjern beskjeden
      </span>
    </button>
  </div>
</div>
`;

exports[`Beskjed with null as link 1`] = `
<div
  className="panel beskjed panel--border"
>
  <div
    className="beskjed__ikon"
  >
    <svg
      fill="none"
      height="48"
      viewBox="0 0 54 54"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="27"
        cy="27"
        fill="#97E6FF"
        r="27"
      />
      <path
        clipRule="evenodd"
        d="M15 27C15 20.3833 20.3833 15 27 15C33.6167 15 39 20.3833 39 27C39 33.6167 33.6167 39 27 39C20.3833 39 15 33.6167 15 27ZM16.0435 27C16.0435 33.0417 20.9583 37.9565 27 37.9565C33.0417 37.9565 37.9565 33.0417 37.9565 27C37.9565 20.9582 33.0417 16.0435 27 16.0435C20.9583 16.0435 16.0435 20.9582 16.0435 27ZM30.1304 33.7824H27.5217V24.9129C27.5217 24.6249 27.288 24.3911 27 24.3911H24.913C24.625 24.3911 24.3913 24.6249 24.3913 24.9129C24.3913 25.2009 24.625 25.4346 24.913 25.4346H26.4783V33.7824H23.8696C23.5816 33.7824 23.3478 34.0162 23.3478 34.3042C23.3478 34.5922 23.5816 34.8259 23.8696 34.8259H30.1304C30.4184 34.8259 30.6522 34.5922 30.6522 34.3042C30.6522 34.0162 30.4184 33.7824 30.1304 33.7824ZM27.5217 20.7393C27.5217 21.3156 27.0546 21.7828 26.4783 21.7828C25.902 21.7828 25.4348 21.3156 25.4348 20.7393C25.4348 20.163 25.902 19.6958 26.4783 19.6958C27.0546 19.6958 27.5217 20.163 27.5217 20.7393Z"
        fill="#262626"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="beskjed__tekst"
  >
    <p
      className="typo-normal"
    >
      <span>
        Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepen... 
        
        <span>
          
          <a
            className="lenke utvid-tekst-lenke__aapne"
            href="#"
            onClick={[Function]}
          >
            Åpne
            <span
              className="nav-frontend-chevron chevronboks chevron--ned"
            />
          </a>
        </span>
      </span>
    </p>
    <p
      className="typo-undertekst beskjed__etikett"
    >
      13.03.2020 - 09:53
    </p>
  </div>
  <div
    className="beskjed__knapp"
  >
    <button
      className="lukknapp lukknapp--bla"
      onClick={[Function]}
    >
      <span
        className="text-hide"
      >
        Fjern beskjeden
      </span>
    </button>
  </div>
</div>
`;

exports[`Beskjed with sikkerhetsnivaa 3 and innloggingsnivaa 3 1`] = `
<div
  className="panel beskjed panel--border"
>
  <div
    className="beskjed__ikon"
  >
    <svg
      fill="none"
      height="48"
      viewBox="0 0 54 54"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="27"
        cy="27"
        fill="#97E6FF"
        r="27"
      />
      <path
        clipRule="evenodd"
        d="M15 27C15 20.3833 20.3833 15 27 15C33.6167 15 39 20.3833 39 27C39 33.6167 33.6167 39 27 39C20.3833 39 15 33.6167 15 27ZM16.0435 27C16.0435 33.0417 20.9583 37.9565 27 37.9565C33.0417 37.9565 37.9565 33.0417 37.9565 27C37.9565 20.9582 33.0417 16.0435 27 16.0435C20.9583 16.0435 16.0435 20.9582 16.0435 27ZM30.1304 33.7824H27.5217V24.9129C27.5217 24.6249 27.288 24.3911 27 24.3911H24.913C24.625 24.3911 24.3913 24.6249 24.3913 24.9129C24.3913 25.2009 24.625 25.4346 24.913 25.4346H26.4783V33.7824H23.8696C23.5816 33.7824 23.3478 34.0162 23.3478 34.3042C23.3478 34.5922 23.5816 34.8259 23.8696 34.8259H30.1304C30.4184 34.8259 30.6522 34.5922 30.6522 34.3042C30.6522 34.0162 30.4184 33.7824 30.1304 33.7824ZM27.5217 20.7393C27.5217 21.3156 27.0546 21.7828 26.4783 21.7828C25.902 21.7828 25.4348 21.3156 25.4348 20.7393C25.4348 20.163 25.902 19.6958 26.4783 19.6958C27.0546 19.6958 27.5217 20.163 27.5217 20.7393Z"
        fill="#262626"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="beskjed__tekst"
  >
    <p
      className="typo-normal"
    >
      <span>
        Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepen... 
        
        <span>
          
          <a
            className="lenke utvid-tekst-lenke__aapne"
            href="#"
            onClick={[Function]}
          >
            Åpne
            <span
              className="nav-frontend-chevron chevronboks chevron--ned"
            />
          </a>
        </span>
      </span>
    </p>
    <p
      className="typo-undertekst beskjed__etikett"
    >
      13.03.2020 - 09:53
    </p>
  </div>
  <div
    className="beskjed__knapp"
  >
    <button
      className="lukknapp lukknapp--bla"
      onClick={[Function]}
    >
      <span
        className="text-hide"
      >
        Fjern beskjeden
      </span>
    </button>
  </div>
</div>
`;

exports[`Beskjed with sikkerhetsnivaa 3 and innloggingsnivaa 4 1`] = `
<div
  className="panel beskjed panel--border"
>
  <div
    className="beskjed__ikon"
  >
    <svg
      fill="none"
      height="48"
      viewBox="0 0 54 54"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="27"
        cy="27"
        fill="#97E6FF"
        r="27"
      />
      <path
        clipRule="evenodd"
        d="M15 27C15 20.3833 20.3833 15 27 15C33.6167 15 39 20.3833 39 27C39 33.6167 33.6167 39 27 39C20.3833 39 15 33.6167 15 27ZM16.0435 27C16.0435 33.0417 20.9583 37.9565 27 37.9565C33.0417 37.9565 37.9565 33.0417 37.9565 27C37.9565 20.9582 33.0417 16.0435 27 16.0435C20.9583 16.0435 16.0435 20.9582 16.0435 27ZM30.1304 33.7824H27.5217V24.9129C27.5217 24.6249 27.288 24.3911 27 24.3911H24.913C24.625 24.3911 24.3913 24.6249 24.3913 24.9129C24.3913 25.2009 24.625 25.4346 24.913 25.4346H26.4783V33.7824H23.8696C23.5816 33.7824 23.3478 34.0162 23.3478 34.3042C23.3478 34.5922 23.5816 34.8259 23.8696 34.8259H30.1304C30.4184 34.8259 30.6522 34.5922 30.6522 34.3042C30.6522 34.0162 30.4184 33.7824 30.1304 33.7824ZM27.5217 20.7393C27.5217 21.3156 27.0546 21.7828 26.4783 21.7828C25.902 21.7828 25.4348 21.3156 25.4348 20.7393C25.4348 20.163 25.902 19.6958 26.4783 19.6958C27.0546 19.6958 27.5217 20.163 27.5217 20.7393Z"
        fill="#262626"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="beskjed__tekst"
  >
    <p
      className="typo-normal"
    >
      <span>
        Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepen... 
        
        <span>
          
          <a
            className="lenke utvid-tekst-lenke__aapne"
            href="#"
            onClick={[Function]}
          >
            Åpne
            <span
              className="nav-frontend-chevron chevronboks chevron--ned"
            />
          </a>
        </span>
      </span>
    </p>
    <p
      className="typo-undertekst beskjed__etikett"
    >
      13.03.2020 - 09:53
    </p>
  </div>
  <div
    className="beskjed__knapp"
  >
    <button
      className="lukknapp lukknapp--bla"
      onClick={[Function]}
    >
      <span
        className="text-hide"
      >
        Fjern beskjeden
      </span>
    </button>
  </div>
</div>
`;

exports[`Beskjed with sikkerhetsnivaa 4 and innloggingsnivaa 3 1`] = `
<div
  className="panel beskjed panel--border"
>
  <div
    className="beskjed__ikon"
  >
    <svg
      fill="none"
      height="48"
      viewBox="0 0 54 54"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="27"
        cy="27"
        fill="#97E6FF"
        r="27"
      />
      <path
        clipRule="evenodd"
        d="M15 27C15 20.3833 20.3833 15 27 15C33.6167 15 39 20.3833 39 27C39 33.6167 33.6167 39 27 39C20.3833 39 15 33.6167 15 27ZM16.0435 27C16.0435 33.0417 20.9583 37.9565 27 37.9565C33.0417 37.9565 37.9565 33.0417 37.9565 27C37.9565 20.9582 33.0417 16.0435 27 16.0435C20.9583 16.0435 16.0435 20.9582 16.0435 27ZM30.1304 33.7824H27.5217V24.9129C27.5217 24.6249 27.288 24.3911 27 24.3911H24.913C24.625 24.3911 24.3913 24.6249 24.3913 24.9129C24.3913 25.2009 24.625 25.4346 24.913 25.4346H26.4783V33.7824H23.8696C23.5816 33.7824 23.3478 34.0162 23.3478 34.3042C23.3478 34.5922 23.5816 34.8259 23.8696 34.8259H30.1304C30.4184 34.8259 30.6522 34.5922 30.6522 34.3042C30.6522 34.0162 30.4184 33.7824 30.1304 33.7824ZM27.5217 20.7393C27.5217 21.3156 27.0546 21.7828 26.4783 21.7828C25.902 21.7828 25.4348 21.3156 25.4348 20.7393C25.4348 20.163 25.902 19.6958 26.4783 19.6958C27.0546 19.6958 27.5217 20.163 27.5217 20.7393Z"
        fill="#262626"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="beskjed__tekst"
  >
    <p
      className="typo-normal"
    >
      <span>
        Du har fått en beskjed.
        <a
          className="lenke panel-lenke"
          href="undefined"
          id="panel-lenke-id"
          onClick={[Function]}
        >
          Logg inn med høyere sikkerhetsnivå for å se beskjeden
        </a>
      </span>
    </p>
    <p
      className="typo-undertekst beskjed__etikett"
    >
      13.03.2020 - 09:53
    </p>
  </div>
</div>
`;

exports[`Beskjed with sikkerhetsnivaa 4 and innloggingsnivaa 4 1`] = `
<div
  className="panel beskjed panel--border"
>
  <div
    className="beskjed__ikon"
  >
    <svg
      fill="none"
      height="48"
      viewBox="0 0 54 54"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="27"
        cy="27"
        fill="#97E6FF"
        r="27"
      />
      <path
        clipRule="evenodd"
        d="M15 27C15 20.3833 20.3833 15 27 15C33.6167 15 39 20.3833 39 27C39 33.6167 33.6167 39 27 39C20.3833 39 15 33.6167 15 27ZM16.0435 27C16.0435 33.0417 20.9583 37.9565 27 37.9565C33.0417 37.9565 37.9565 33.0417 37.9565 27C37.9565 20.9582 33.0417 16.0435 27 16.0435C20.9583 16.0435 16.0435 20.9582 16.0435 27ZM30.1304 33.7824H27.5217V24.9129C27.5217 24.6249 27.288 24.3911 27 24.3911H24.913C24.625 24.3911 24.3913 24.6249 24.3913 24.9129C24.3913 25.2009 24.625 25.4346 24.913 25.4346H26.4783V33.7824H23.8696C23.5816 33.7824 23.3478 34.0162 23.3478 34.3042C23.3478 34.5922 23.5816 34.8259 23.8696 34.8259H30.1304C30.4184 34.8259 30.6522 34.5922 30.6522 34.3042C30.6522 34.0162 30.4184 33.7824 30.1304 33.7824ZM27.5217 20.7393C27.5217 21.3156 27.0546 21.7828 26.4783 21.7828C25.902 21.7828 25.4348 21.3156 25.4348 20.7393C25.4348 20.163 25.902 19.6958 26.4783 19.6958C27.0546 19.6958 27.5217 20.163 27.5217 20.7393Z"
        fill="#262626"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="beskjed__tekst"
  >
    <p
      className="typo-normal"
    >
      <span>
        Vi mottok søknaden din 18. september 2019. Du kan følge med på statusen i Dine foreldrepen... 
        
        <span>
          
          <a
            className="lenke utvid-tekst-lenke__aapne"
            href="#"
            onClick={[Function]}
          >
            Åpne
            <span
              className="nav-frontend-chevron chevronboks chevron--ned"
            />
          </a>
        </span>
      </span>
    </p>
    <p
      className="typo-undertekst beskjed__etikett"
    >
      13.03.2020 - 09:53
    </p>
  </div>
  <div
    className="beskjed__knapp"
  >
    <button
      className="lukknapp lukknapp--bla"
      onClick={[Function]}
    >
      <span
        className="text-hide"
      >
        Fjern beskjeden
      </span>
    </button>
  </div>
</div>
`;
